include/crow/app.h

```diff
@@ -77,13 +77,17 @@ namespace crow
             return router_.new_rule_dynamic(std::move(rule));
         }
 
-        /// Create a route using a rule (**Use CROW_ROUTE instead**)
+        ///Create a route using a rule (**Use CROW_ROUTE instead**)
         template<uint64_t Tag>
+#ifdef CROW_GCC83_WORKAROUND
+        auto& route(std::string&& rule)
+#else
         auto route(std::string&& rule)
 #ifdef CROW_CAN_USE_CPP17
           -> typename std::invoke_result<decltype(&Router::new_rule_tagged<Tag>), Router, std::string&&>::type
 #else
           -> typename std::result_of<decltype (&Router::new_rule_tagged<Tag>)(Router, std::string&&)>::type
+#endif
 #endif
```

Comment:
I figured it out, the problem is that the trailing type needs to be removed, here's my code that worked:
```suggestion
#ifdef CROW_GCC83_WORKAROUND
        auto& route(std::string&& rule)
#else
        auto route(std::string&& rule)
#endif
#if defined CROW_CAN_USE_CPP17 && !defined CROW_GCC83_WORKAROUND
          -> typename std::invoke_result<decltype(&Router::new_rule_tagged<Tag>), Router, std::string&&>::type
#elif !defined CROW_GCC83_WORKAROUND
          -> typename std::result_of<decltype (&Router::new_rule_tagged<Tag>)(Router, std::string&&)>::type
#endif
```

Reply:
Thanks! Sorry I was very busy and did not have time to investigate this myself.